### DISCUSSION TITLE
SMB share server

### DISCUSSION THREAD

**Discussion by ghost** [linuxdeepin/developer-center#3231] [Features Request & Ideas | 特性请求 & 头脑风暴]
Hi,
SMB file share from a local network is very slow and takes a lot of time every time reload from the network storage to the PC. I would suggest to keep a local copy on user PC of files from network attached storage and do like changes sync. There is a real problem when files are big
![Screenshot_dde-file-manager_20220607093736](https://user-images.githubusercontent.com/98664445/172313571-06ece1f6-8ee0-4019-bff3-e3ad159dd8eb.png)

**Comment by BLumia** [comment]
IMO, I think it might be a bad idea. For example, the deepin team internally share testing resources and ISOs via samba share, which means the files can be large. In this case, downloading those files by default will:

1. download a lot of files to local storage and consume your free disk spaces
2. hog the network and server resources on the samba server
3. since samba scheme is not something like rsync, we probably cannot know if a file need to be re-downloaded

Also, in fact a samba share is not that slow if it's on the local network (which is the most frequently used use-case).

**Reply by ghost** [reply]
For some reason shared folder works really slow, It stays like frozen when I want to upload a document file up to 2 minutes or more till process changes and refresh with file uploaded

**Reply by BLumia** [reply]
You can also check if other file manager apps (like `nautilus`) are also slow, if it's not, then it's probably a performance issue of dde-file-manager.

**Reply by ghost** [reply]
> You can also check if other file manager apps (like `nautilus`) are also slow, if it's not, then it's probably a performance issue of dde-file-manager.

Thank you :) Nautilus better performance
